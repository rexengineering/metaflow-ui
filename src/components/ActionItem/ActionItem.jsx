import PropTypes from "prop-types";
import {
  Checkbox,
  IconButton,
  makeStyles,
  Typography,
} from "@material-ui/core";
import React from "react";
import { faTrashAlt } from "@fortawesome/pro-light-svg-icons/faTrashAlt";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import clsx from "clsx";
import {
  ActionItemCheckType,
  ActionItemDeleteType,
  ActionItemDisplayOnlyType,
} from "../../constants";

const useStyles = makeStyles((theme) => ({
  container: {
    display: "flex",
    alignItems: "center",
    padding: theme.spacing(1, 2),
    background: theme.palette.grey["600"],
    borderRadius: theme.spacing(1),
    marginBottom: theme.spacing(1.5),
  },
  label: {
    flexGrow: 1,
    color: theme.palette.grey["200"],
  },
  actionElement: {
    display: "inline-block",
    background: theme.palette.grey["800"],
    color: theme.palette.common.white,
    borderRadius: theme.spacing(1),
  },
  iconButtonSize: {
    fontSize: theme.spacing(2.5),
  },
}));

function ActionItem({ actionType, label, onAction, isChecked, className }) {
  const classes = useStyles();
  const handleOnAction = () => onAction(label, actionType);
  return (
    <div className={clsx(classes.container, className)}>
      <Typography className={classes.label}>{label}</Typography>
      <div className={classes.actionElement}>
        {
          {
            [ActionItemDisplayOnlyType]: null,
            [ActionItemCheckType]: (
              <Checkbox checked={isChecked} onChange={handleOnAction} />
            ),
            [ActionItemDeleteType]: (
              <IconButton
                onClick={handleOnAction}
                className={classes.iconButtonSize}
              >
                <FontAwesomeIcon icon={faTrashAlt} />
              </IconButton>
            ),
          }[actionType]
        }
      </div>
    </div>
  );
}

ActionItem.defaultProps = {
  onAction: () => {},
  isChecked: false,
  className: "",
};

ActionItem.propTypes = {
  actionType: PropTypes.oneOf([
    ActionItemCheckType,
    ActionItemDeleteType,
    ActionItemDisplayOnlyType,
  ]).isRequired,
  label: PropTypes.string.isRequired,
  onAction: PropTypes.func,
  isChecked: PropTypes.bool,
  className: PropTypes.string,
};

export default ActionItem;
