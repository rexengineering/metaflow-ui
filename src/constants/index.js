export const BUSY = "BUSY";
export const OFFLINE = "OFFLINE";
export const ONLINE = "ONLINE";

export const ActionItemDeleteType = "ActionDeleteType";
export const ActionItemCheckType = "ActionCheckType";
export const ActionItemDisplayOnlyType = "ActionDisplayOnlyType";

export const InitialButtonState = 'initial';
export const LoadingButtonState = 'loading';
export const talkTrackIdentifierProp = 'iid';
export const talkTrackType = 'talktrack';