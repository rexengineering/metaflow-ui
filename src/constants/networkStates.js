export const INITIAL = "INITIAL";
export const REQUEST = "REQUEST";
export const SUCCESS = "SUCCESS";
export const FAILURE = "FAILURE";

export default {
  INITIAL,
  REQUEST,
  SUCCESS,
  FAILURE
}
