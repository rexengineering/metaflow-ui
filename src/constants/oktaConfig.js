const config = {
  oidc: {
    clientId: window.REX.OKTA.CLIENT_ID,
    issuer: window.REX.OKTA.ISSUER,
    redirectUri: window.REX.OKTA.REDIRECT_URI,
    scopes: ["openid", "profile", "email"],
    pkce: true,
    disableHttpsCheck: window.REX.OKTA.OKTA_TESTING_DISABLEHTTPSCHECK,
  },
  resourceServer: {
    messagesUrl: "http://localhost:5000/api/messages",
  },
};

export default config;
