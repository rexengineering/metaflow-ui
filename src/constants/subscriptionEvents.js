export const START_WORKFLOW = "START_WORKFLOW";
export const UPDATE_WORKFLOW = "UPDATE_WORKFLOW";
export const FINISH_WORKFLOW = "FINISH_WORKFLOW";
export const START_TASK = "START_TASK";
export const UPDATE_TASK = "UPDATE_TASK";
export const FINISH_TASK = "FINISH_TASK";
export const START_BROADCAST = "START_BROADCAST";
export const FINISH_BROADCAST = "FINISH_BROADCAST";
export const ERROR_BROADCAST = "ERROR_BROADCAST";
export const KEEP_ALIVE_SUCCESS = "SUCCESS";
export const KEEP_ALIVE_FAILURE = "FAILURE";