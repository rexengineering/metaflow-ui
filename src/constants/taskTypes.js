import { string, number, boolean } from "yup";

// Inputs
export const TEXT = "TEXT";
export const PHONE_NUMBER = "PHONE_NUMBER";
export const PERCENTAGE = "PERCENTAGE";
export const CURRENCY = "CURRENCY";
export const BOOLEAN = "BOOLEAN";
export const INTEGER = "INTEGER";
export const FLOAT = "FLOAT";

export function isInputType(field) {
  return [
    TEXT,
    PHONE_NUMBER,
    PERCENTAGE,
    CURRENCY,
    BOOLEAN,
    INTEGER,
    FLOAT,
  ].includes(field);
}

// Info
export const TABLE = "TABLE";
export const COPY = "COPY";
export const WORKFLOW_INSTANTIATOR = "WORKFLOW";

export function isInfoType(type) {
  return [TABLE, COPY, WORKFLOW_INSTANTIATOR].includes(type);
}

const numberValidationMessage = "Please enter a number";

export const validationSchemaMapping = {
  TEXT: string("Please enter a value"),
  PHONE_NUMBER: string("Please enter a value").length(10),
  PERCENTAGE: number(numberValidationMessage).typeError(
    numberValidationMessage
  ),
  CURRENCY: number(numberValidationMessage).typeError(numberValidationMessage),
  BOOLEAN: boolean(),
  INTEGER: number(numberValidationMessage)
    .typeError(numberValidationMessage)
    .integer(),
  FLOAT: number(numberValidationMessage).typeError(numberValidationMessage),
};

const taskTypes = {
  TEXT,
  PHONE_NUMBER,
  PERCENTAGE,
  CURRENCY,
  BOOLEAN,
  INTEGER,
  FLOAT,
  TABLE,
  COPY,
  WORKFLOW_INSTANTIATOR,
};

export default taskTypes;
