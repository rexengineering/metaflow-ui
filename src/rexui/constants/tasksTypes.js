import { string, number, boolean } from "yup";


const numberValidationMessage = "Please enter a number";

export const TEXT = "TEXT";
export const PHONE_NUMBER = "PHONE_NUMBER";
export const PERCENTAGE = "PERCENTAGE";
export const CURRENCY = "CURRENCY";
export const BOOLEAN = "BOOLEAN";
export const INTEGER = "INTEGER";
export const FLOAT = "FLOAT";
export const TABLE = "TABLE";
export const COPY = "COPY";
export const WORKFLOW_INSTANTIATOR = "WORKFLOW";

export const validationSchemaMapping = {
    TEXT: string("Please enter a value"),
    PHONE_NUMBER: string("Please enter a value").length(10),
    PERCENTAGE: number(numberValidationMessage).typeError(numberValidationMessage),
    CURRENCY: number(numberValidationMessage).typeError(numberValidationMessage),
    BOOLEAN: boolean(),
    INTEGER: number(numberValidationMessage)
        .typeError(numberValidationMessage)
        .integer(),
    FLOAT: number(numberValidationMessage).typeError(numberValidationMessage),
};

export function isInfoType(type) {
    return [TABLE, COPY, WORKFLOW_INSTANTIATOR].includes(type);
}

export function isInputType(field) {
    return [
        TEXT,
        PHONE_NUMBER,
        PERCENTAGE,
        CURRENCY,
        BOOLEAN,
        INTEGER,
        FLOAT,
    ].includes(field);
}