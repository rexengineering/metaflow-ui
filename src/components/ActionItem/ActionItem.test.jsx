import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import { ActionItemDeleteType } from "../../constants";
import ActionItem from "./index";

describe("<ActionItem />", () => {
  const renderActionItemComponent = (
    label = "Direct Buyer",
    actionType = ActionItemDeleteType,
    onAction = () => {}
  ) => <ActionItem label={label} actionType={actionType} onAction={onAction} />;

  it("should render", () => {
    const { asFragment } = render(renderActionItemComponent());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderActionItemComponent());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render provided label", () => {
    const label = "New house!";
    render(renderActionItemComponent(label));
    const labelNode = screen.getByText(label);
    expect(labelNode).toBeInTheDocument();
  });

  it("should handle onAction properly", () => {
    const onAction = jest.fn();
    render(
      renderActionItemComponent("New house", ActionItemDeleteType, onAction)
    );
    const buttonNode = screen.getByRole("button");
    fireEvent.click(buttonNode);
    expect(onAction).toBeCalled();
  });
});
