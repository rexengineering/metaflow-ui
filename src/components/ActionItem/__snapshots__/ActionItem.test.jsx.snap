// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionItem /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-5"
  >
    <p
      class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
    >
      Direct Buyer
    </p>
    <div
      class="makeStyles-actionElement-7"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButtonSize-8"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash-alt fa-w-14 "
            data-icon="trash-alt"
            data-prefix="fal"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 432h16a8 8 0 0 0 8-8V152a8 8 0 0 0-8-8h-16a8 8 0 0 0-8 8v272a8 8 0 0 0 8 8zm-160 0h16a8 8 0 0 0 8-8V152a8 8 0 0 0-8-8h-16a8 8 0 0 0-8 8v272a8 8 0 0 0 8 8zM440 64H336l-33.6-44.8A48 48 0 0 0 264 0h-80a48 48 0 0 0-38.4 19.2L112 64H8a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8h24v368a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V96h24a8 8 0 0 0 8-8V72a8 8 0 0 0-8-8zM171.2 38.4A16.1 16.1 0 0 1 184 32h80a16.1 16.1 0 0 1 12.8 6.4L296 64H152zM384 464a16 16 0 0 1-16 16H80a16 16 0 0 1-16-16V96h320zm-168-32h16a8 8 0 0 0 8-8V152a8 8 0 0 0-8-8h-16a8 8 0 0 0-8 8v272a8 8 0 0 0 8 8z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
